Src/Plugins/AzureDevOpsConfiguration/ServiceConnectionSecurityValidators/AZC102_190.DoNotGrantAllPipelinesAccessValidator.cs

```diff
@@ -75,11 +74,13 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary
                                                                 IDictionary<string, string> options,
                                                                 ref ResultLevelKind resultLevelKind)
         {
+            string asset = string.Empty;
             try
             {
                 string organization = fingerprint.Host;
                 string project = fingerprint.Resource;
                 string serviceConnectionId = fingerprint.Id;
+                asset = $"https://{organization}/{project}/_apis/pipelines/pipelinePermissions/endpoint/{serviceConnectionId}";
```

Comment:
>asset [](http://example.com/codeflow?start=16&length=5)

Here's a good example of what we want, an asset id that shows the compromised thing, in this case, an ADO service connection. If we don't have that, we'll use the truncated secret to provide uniqueness in the output.